rust/src/passwd.rs

```diff
@@ -404,24 +403,19 @@ fn concat_files(rootfs: &openat::Dir, prev_root: &gio::File, target: &str) -> Re
         let usrlib_src = format!("usr/lib/{}", &target);
         prev_root.resolve_relative_path(usrlib_src)
     };
-    if orig_usretc_content.is_none() && orig_usrlib_content.is_none() {
```

Comment:
Did this get drop by mistake? I think we aren't currently exercising this, so it does not have any actual visible effects, but the existing logic seems correct to me and it may just need porting?

Reply:
When I was updating this my thought was the old code was buggy and we should have been checking for file existence from the start.  But indeed, I glossed over this and it needs investigation.

Reply:
OK yeah I looked, the old code was busted here.  https://github.com/ostreedev/ostree/blob/5023278293d20fe70f61d89f01d6356fa5ef2ffc/src/libostree/ostree-repo-file.c#L619 never returns `NULL`, and neither should any other `GFile` implementations.  So the nullability was fixed, and this code always wanted to be checking for file existence.

Reply:
Oh but we do need to check for existence here still